odoo/addons/base/views/res_partner_views.xml

```diff
@@ -81,11 +81,12 @@
                     <div class="oe_title">
                         <field name="company_type" options="{'horizontal': true}" widget="radio" groups="base.group_no_one"/>
                         <h1>
-                            <field name="name" default_focus="1" placeholder="e.g. John Doe" attrs="{'required' : [('type', '=', 'contact')]}"/>
+                            <field name="name" default_focus="1" placeholder="e.g. Lumber Inc" attrs="{'required' : [('type', '=', 'contact')], 'invisible': [('is_company','=', False)]}"/>
+                            <field name="name" default_focus="1" placeholder="e.g. Brandom Freeman" attrs="{'required' : [('type', '=', 'contact')], 'invisible': [('is_company','=', True)]}"/>
```

Comment:
@ged-odoo does having the same field twice in a view now work as long as they're not visible at the same time? Or is this going to break because the form will plug in on the wrong one?

Reply:
It's fine as long as `required` is the same
https://www.odoo.com/documentation/14.0/reference/views.html#semantic-components
(I asked (GED) to be sure ;D)